src/sidebar/permissions.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+var STORAGE_KEY = 'hypothesis.privacy';
+
+/**
+ * Object defining which principals can read, update and delete an annotation.
+ *
+ * This is the same as the `permissions` field retrieved on an annotation via
+ * the API.
+ *
+ * Principals are strings of the form `type:id` where `type` is `'acct'` (for a
+ * specific user) or `'group'` (for a group).
+ *
+ * @typedef Permissions
+ * @property {string[]} read - List of principals that can read the annotation
+ * @property {string[]} update - List of principals that can edit the annotation
+ * @property {string[]} delete - List of principals that can delete the
+ * annotation
+ */
+
+/**
+ * A service for generating and querying `Permissions` objects for annotations.
+ *
+ * It also provides methods to save and restore permissions preferences for new
+ * annotations to local storage.
+ */
+// @ngInject
+function Permissions(localStorage) {
```

Comment:
I think it's a little confusing that bot the service and the typedef above are called `Permissions`.